engine/api/src/main/java/io/deephaven/engine/table/Table.java

```diff
@@ -1761,6 +1761,23 @@ Table updateBy(@NotNull final UpdateByControl control,
     Table updateBy(@NotNull final UpdateByControl control,
             @NotNull final Collection<? extends UpdateByClause> operations);
 
+    /**
+     * <p>
+     * Create a table with the same {@link #getRowSet() rowSet} as its parent that will perform the specified set of row
+     * based operations to it. As opposed to {@link #update(String...)} these operations are more restricted but are
+     * capable of processing state between rows.This operation will group the table by the specified set of keys if
+     * provided
+     * </p>
+     *
+     * @param operations the operations to apply to the table.
+     * @param byColumns the columns to group by before applying.
+     * @return a table with the same index, with the specified operations applied to each group defined by the
+     *         {@code byColumns}
+     */
+    @ConcurrentMethod
+    Table updateBy(@NotNull Collection<? extends UpdateByClause> operations,
+            @NotNull Collection<? extends Selectable> byColumns);
```

Comment:
Note: I left this out of your original review (b/c it was easy for me to add in this PR); but from a QST point of view, I need to be able to call updateBy w/ these exact parameters. Usage in TableAdapterImpl.